src/Dapr.Client/Extensions/HttpExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2024 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Dapr.Client
+{
+    /// <summary>
+    /// Provides extensions specific to HTTP types.
+    /// </summary>
+    internal static class HttpExtensions
+    {
+        /// <summary>
+        /// Appends key/value pairs to the query string on an HttpRequestMessage.
+        /// </summary>
+        /// <param name="uri">The uri to append the query string parameters to.</param>
+        /// <param name="queryStringParameters">The key/value pairs to populate the query string with.</param>
+        public static Uri AddQueryParameters(this Uri? uri,
+            IReadOnlyCollection<KeyValuePair<string, string>>? queryStringParameters)
```

Comment:
Curious why not [NameValueCollection](https://learn.microsoft.com/en-us/dotnet/api/system.collections.specialized.namevaluecollection?view=net-8.0#remarks)?

Reply:
Fundamentally, I was looking to ensure that a read-only collection is passed. Conceptually, they wind up operationally working the same way. I don't see why an overload couldn't be added for a `NameValueCollection` as well.